contributors/design-proposals/kubeadm-upgrades.md

```diff
@@ -0,0 +1,350 @@
+ - **Required upgrade policy**: An upgrade policy that must be satisfied; a precondition the user can't ignore.
+ - **Skippable upgrade policy**: An upgrade policy that should be satisfied, but is skippable by passing the `--force` flag.
+ - **Runtime reconfiguration**: An user invokes the `kubeadm upgrade apply` command with the current version, but with a different configuration.
+
+### Pre-upgrade policies
+
+ - The kubeadm CLI version must be higher or equal to the upgrade target version
+   - **Required policy**: The kubeadm CLI minor version must be equal to or higher than the upgrade target minor version.
+     - Example: kubeadm CLI version `v1.8.3` can upgrade a cluster to `v1.7.4` and `v1.8.5`, but not `v1.9.0`.
+   - **Optional policy**: The kubeadm CLI patch version should be equal to or higher than the upgrade target patch version.
+     - Example: kubeadm CLI version `v1.8.3` can upgrade a cluster to `v1.8.3` without an explicit grant, but upgrading to `v1.8.5` requires forcing. 
+ - The kubeadm CLI of version `vX.Y.Z` can create clusters with minor versions `Y` and `Y-1`.
+   - This is a **required policy**, and is not specific to upgrades. This policy is also is also enforced when creating a cluster.
+     - Example: kubeadm CLI version `v1.8.2` can create clusters of (and upgrade clusters to) version `v1.8.x` and `v1.7.x`.
+ - The control plane version must always be higher than or equal to the version of all the kubelets, and the maximum allowed skew is one minor versions.
+   - **Required policy**: The control plane version must be higher than or equal to the kubelet version. This is a generic Kubernetes policy.
+     - Example: For a control plane of version `v1.8.4`; a kubelet version may be `v1.8.2`, but not `v1.9.3`
+   - **Skippable policy**: The maximum allowed minor skew between the control plane and the kubelets is one minor version.
+     - Example: For a control plane of version `v1.7.2` being upgraded to `v1.8.5`; the minimum allowed kubelet version is `v1.7.0`.
+     - Note: We might relax this policy to allow two minor versions before going to GA.
+ - The version change must be positive, i.e. only upgrades are supported; not downgrades or runtime reconfiguration.
+   - This is a **skippable policy** in v1.8, and will be removed in future versions before going to GA.
+     - Example: Downgrading from `v1.8.3` to `v1.8.2` or reconfiguring a cluster on version `v1.8.1` can only be done (initially) along with the `--force` flag.
+ - The maximum allowed upgrade minor skew is one minor version.
+   - This is a **required policy**. You can upgrade from `vX.Y.Z1` to `vX.Y+1.Z2`, but not to `vX.Y+2.Z3`.
+     - Example: Upgrading from `v1.6.2` to `v1.8.4` is not supported.
+     - Note: We might relax this policy to allow two minor versions before going to GA.
```

Comment:
this seems reasonable for the control plane. once we start talking about kubelet upgrades i think we will want to support upgrading multiple minor versions at a time.

Reply:
yeah, this focuses on the control plane for now. Let's see what we do with the kubelets here later